examples/pipeline_wavernn/losses.py

```diff
@@ -0,0 +1,90 @@
+import torch
+from torch.nn import functional as F
+
+
+def log_sum_exp(x):
+    r""" Numerically stable log_sum_exp implementation that prevents overflow
+    """
+
+    axis = len(x.size()) - 1
+    m, _ = torch.max(x, dim=axis)
+    m2, _ = torch.max(x, dim=axis, keepdim=True)
+    return m + torch.log(torch.sum(torch.exp(x - m2), dim=axis))
+
+
+def MoLLoss(y_hat, y, num_classes=65536, log_scale_min=None, reduce=True):
```

Comment:
nit: `mol_loss` given that this is not a class

Reply:
Yes, I have updated it.